server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java

```diff
@@ -368,5 +396,66 @@ private PhysicalPlan transformQuery(QueryOperator queryOperator)
     return new ArrayList<>(columnList.subList(seriesOffset, endPosition));
   }
 
+
+  private void deduplicateAggregation(QueryPlan queryPlan)
+      throws QueryProcessException {
+    List<Path> paths = queryPlan.getPaths();
+    List<String> aggregations = queryPlan.getAggregations();
+
+    Map<Path, TSDataType> dataTypeMapping = queryPlan.getDataTypeMapping();
+    if (paths == null) {
```

Comment:
This could also be removed.
If the user selects some non-existing paths, the ConcatPathOptimizer will throw an exception.